private-templates-service/client/src/store/currentTemplate/types.ts

```diff
@@ -0,0 +1,34 @@
+import { IncomingMessage } from "http";
+
+export interface CurrentTemplateState {
+  templateID: string | undefined;
```

Comment:
Would it make more sense to make these properties optional instead of 'undefined'? Eg:
`templateID?: string`

Reply:
I made it undefined because the whole `currentTemplate` store is irrelevant if the user is not on the Designer page. So in `currentTemplate/reducers.ts` I set the initialState to be such that all these variables are set to undefined. If I make it optional there is a scenario where the user can navigate to the Dashboard but the redux store reflects that there is a template currently being displayed.

Reply:
Under the hood, `templateID?: string` == ` templateID: string | undefined;`

Reply:
Oh okay, will change